src/terrain/static_mesh.cpp

```diff
@@ -10,10 +11,35 @@
 
 namespace terrain {
 
+StaticMesh::StaticMesh(entity::Mesh mesh,
+                       const std::vector<glm::vec3>& model_transforms) {
+    init(mesh.indices_,
+         mesh.indexed_vertices_,
+         mesh.indexed_colors_,
+         mesh.indexed_normals_,
+         model_transforms);
+}
+
 StaticMesh::StaticMesh(
-    std::vector<unsigned short>& indices, std::vector<glm::vec3>& indexed_vertices,
-    std::vector<glm::vec3>& indexed_colors, std::vector<glm::vec3>& indexed_normals,
-    std::vector<glm::vec3>& model_transforms
+    const std::vector<unsigned short>& indices,
+    const std::vector<glm::vec3>& indexed_vertices,
+    const std::vector<glm::vec3>& indexed_colors,
+    const std::vector<glm::vec3>& indexed_normals,
+    const std::vector<glm::vec3>& model_transforms
+) {
+    init(indices,
+         indexed_vertices,
+         indexed_colors,
+         indexed_normals,
+         model_transforms);
+}
+
+void StaticMesh::init(
+    const std::vector<unsigned short>& indices,
+    const std::vector<glm::vec3>& indexed_vertices,
+    const std::vector<glm::vec3>& indexed_colors,
+    const std::vector<glm::vec3>& indexed_normals,
+    const std::vector<glm::vec3>& model_transforms
```

Comment:
Ok this code is much better but why don't you make `init` into a constructor and then fall back to it?

Reply:
nino you are demonstrating you know less cpp then me